app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+			}
+		}
+
+		assert.EqualValues(s.T(),
+			expectedSpendableBal,
+			balances.AmountOf(app.BondDenom).Int64(),
+			"spendable balance must match")
+
+		_, err = cctx.WaitForTimestamp(startTime.Add(periods[i].Duration() + 10*time.Millisecond)) // Wait for the next period to be passed
+		assert.NoError(s.T(), err)
+	}
+
+	s.testTransferVestingAmount(cctx, accName, false)
+	s.testDelegation(cctx, accName)
+	s.testClaimDelegationReward(cctx, accName)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisContinuesVestingAccounts() {
+	// initial unlocked allocation to pay the gas fees
+	initBalanceForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(10))
+
+	startTime := tmtime.Now().Add(10 * time.Second)
+	endTime := startTime.Add(20 * time.Second)
+
+	const accName = "cont_vesting_0"
+	kr := testfactory.GenerateKeyring()
+
+	bacc, coins := testfactory.NewBaseAccount(kr, accName)
+	cva := vestingtypes.NewContinuousVestingAccount(bacc, coins, startTime.Unix(), endTime.Unix())
+
+	gsOpt := func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		vAccs := authtypes.GenesisAccounts{cva}
+		vBals := []banktypes.Balance{
+			{
+				Address: cva.GetAddress().String(),
+				Coins:   cva.GetOriginalVesting().Add(initBalanceForGasFee),
+			},
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	// Test account details correctness
+	acli := authtypes.NewQueryClient(cctx.GRPCClient)
+	res, err := acli.Account(context.Background(), &authtypes.QueryAccountRequest{
+		Address: accAddress,
+	})
+	assert.NoError(s.T(), err)
+
+	var qAcc vestingtypes.ContinuousVestingAccount
+	err = qAcc.Unmarshal(res.Account.Value)
+	assert.NoError(s.T(), err)
+
+	// Checking the queried account data
+	assert.Equal(s.T(), accAddress, qAcc.GetAddress().String(), "account address must match")
+	assert.EqualValues(s.T(), startTime.Unix(), qAcc.GetStartTime(), "start time must match")
+	assert.EqualValues(s.T(), endTime.Unix(), qAcc.GetEndTime(), "end time must match")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+		"original vesting must match")
+
+	/*--------*/
+
+	// Test available balance
+	for i := 0; i < 5; i++ {
+		queryTime := tmtime.Now()
+
+		balances, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+		assert.NoError(s.T(), err)
+
+		minExpectedSpendableBal := cva.GetVestedCoins(queryTime).Add(initBalanceForGasFee)
+
+		assert.LessOrEqual(s.T(),
+			balances.AmountOf(app.BondDenom).Int64(),
+			minExpectedSpendableBal.AmountOf(app.BondDenom).Int64(),
+		)
+
+		_, err = cctx.WaitForTimestamp(startTime.Add(5)) // Wait for a while
```

Comment:
[nit] IMO this line of code is more precise than the comment

```suggestion
		_, err = cctx.WaitForTimestamp(startTime.Add(5))
```